e2e-tests/scheduled-backup/run

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+set -o errexit
+set -o xtrace
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+
+function restore {
+	local cluster=${1}
+	local storage=${2}
+	local backup_type=${3}
+	local restore_target=''
+
+	run_psql \
+		'\c myapp \\\ INSERT INTO myApp (id) VALUES (100501)' \
+		"postgres:postgres_pass@${cluster}.${namespace}"
+
+	compare_psql_cmd \
+		"select-1" \
+		'\c myapp \\\ SELECT * from myApp;' \
+		"postgres:postgres_pass@${cluster}.${namespace}"
+
+	run_psql \
+		'\c myapp \\\ INSERT INTO myApp (id) VALUES (100502)' \
+		"postgres:postgres_pass@${cluster}.${namespace}"
+
+	restore_target="$(run_psql \
+		'\c myapp \\\ SELECT xmin FROM myApp WHERE id='100501';' \
+		"postgres:postgres_pass@${cluster}.${namespace}" | awk '{$1=$1;print}')"
+
+	run_psql \
+		'DROP DATABASE myapp' \
+		"postgres:postgres_pass@${cluster}.${namespace}"
+
+	run_restore ${cluster} ${backup_type} ${storage} "${restore_target}" 'xid'
+
+	compare_psql_cmd \
+		"select-1" \
+		'\c myapp \\\ SELECT * from myApp;' \
+		"postgres:postgres_pass@${cluster}.${namespace}"
+}
+
+function apply_configs {
+	local cluster=${1}
+
+	set +x
+	yq w $test_dir/conf/backrest-repo-secret.yml 'metadata.name' "${cluster}-backrest-repo-config" \
+		| yq w - 'data[aws-s3-key]' "$(yq r -d"0" ${conf_dir}/cloud-secret.yml 'data.AWS_ACCESS_KEY_ID')" \
+		| yq w - 'data[aws-s3-key-secret]' "$(yq r -d"0" ${conf_dir}/cloud-secret.yml 'data.AWS_SECRET_ACCESS_KEY')" \
+		| yq w - 'data[gcs-key]' "$(yq r -d0 ${conf_dir}/cloud-secret-minio-gw.yml 'stringData."credentials.json"' | base64)" \
+		| kubectl_bin apply -f -
+	set -x
+	yq r $test_dir/conf/custom-config.yml \
+		| yq w - 'metadata.name' "${cluster}-custom-config" \
+		| kubectl_bin apply -f -
+
+	kubectl_bin apply \
+		-f $conf_dir/client.yml
+}
+
+function main {
+	create_namespace $namespace
+	deploy_cert_manager
+	deploy_operator
+
+	desc 'create first PG cluster'
+	cluster="some-name-${RANDOM}"
+
+	desc 'Testing backup/restore from s3'
+	apply_configs ${cluster}
+	spinup_pgcluster "${cluster}" "${test_dir}/conf/cr.yaml" "s3" "${cluster}-custom-config"
+	sleep 400
+	restore ${cluster} "s3" "full"
```

Comment:
we need to test all target storage combinations like in existing backup test. Please make sure that full, incr, diff backups are utilized.

Reply:
https://github.com/percona/percona-postgresql-operator/pull/114/commits/df5ca42e0f4239a42855c9ba531d71cee3911a60